azerothcore/azerothcore-wotlk
# fix(DB): Significant Creature improvements in Shadowglen, Teldrassil

## Changes Proposed:
- All creatures in Shadowglen have been overhauled.
- Spawn xyzo values now match data
- Spawn movement values now aproximate based on data.
- Respawns now occur at correct rates.
- A few mobs that should have simple waypoint movements have them now.
- Mobs that should have no movement have been placed correctly.
- A few mobs that should not exist or should not exist at the same time as another mob have been removed, and all spawn points have likely been accounted for.

## SOURCE:

Spawn data gathered from all packet data from area, parsed by WPP+Malcroms tool.
Distance data gathered carefully mostly from waypoint creator tool.  Accuracy varies, but most use a very high quantity of waypoints to solve distance value from.
Some distance data is estimated, and other is rounded, downward.
Rounded and estimated values have less digit precision.
Timers accumulated over time, most of the timers can be viewed here:
https://www.youtube.com/watch?v=xPFt-RpfZ-c

## Tests Performed:

- Ran SQL before creating delete from line for creature table, ensuring no accidental conflicts/typos snuck in, breaking something else while fixing this.
- Checked in game to verify creature locations.


## How to Test the Changes:


1. Create a Night Elf 
2. Compare start area spawns and movement to expected classic look.

## Known Issues and TODO List:


- There is enough data gathered on spawn sharing creatures and this has been written in a way such to easily modify this SQL if proper spawnpoint behavior is implemented.  Certain mobs should share spawn points and those are all noted.  You could implement this via pools, but a more correct method would just have an additional value in the creature table. 

If you have no idea what I'm talking about, see https://www.youtube.com/watch?v=9pwZIaGUWQI


## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

Special thanks to Malcrom, his help saved time and increased accuracy.
Also thanks to Ddraig for finding a specific annoying spawn.
Much love to ymir and WPP devs.
